app/services/authn_context_resolver.rb

```diff
@@ -11,15 +11,32 @@ def initialize(user:, service_provider:, vtr:, acr_values:)
   end
 
   def resolve
-    if vtr.present?
-      selected_vtr_parser_result_from_vtr_list
+    result
+  end
+
+  def asserted_ial_acr
+    return Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF unless user.active_profile.present?
+
+    if result.biometric_comparison?
+      Saml::Idp::Constants::IAL2_BIO_REQUIRED_AUTHN_CONTEXT_CLASSREF
+    elsif result.identity_proofing? ||
+          result.ialmax?
+      Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
     else
-      acr_result
+      Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
     end
   end
 
   private
 
+  def result
```

Comment:
hm i am not a huge fan of creating a new method to pull all of another method's logic into it. what were the considerations of this change?

Reply:
Me neither. I mistakenly assumed that I should use the pattern from the closed PR.

Reply:
Fixed.